aptos-move/framework/aptos-framework/sources/state_storage.move

```diff
@@ -23,59 +21,70 @@ module aptos_framework::state_storage {
         usage: Usage,
     }
 
-    /// This updates at reconfig and guarantees not to change elsewhere, safe
-    /// for gas calculation.
-    ///
-    /// Specifically, it copies the usage at the begining of the concluding
-    /// epoch for gas calculation of the entire next epoch. -- The data is one
-    /// epoch older than ideal, but the Vm doesn't need to worry about reloading
-    /// gas parameters after the first txn of an epoch.
-    struct GasParameter has key, store {
-        usage: Usage,
-    }
-
     public(friend) fun initialize(aptos_framework: &signer) {
         system_addresses::assert_aptos_framework(aptos_framework);
         assert!(
             !exists<StateStorageUsage>(@aptos_framework),
             error::already_exists(ESTATE_STORAGE_USAGE)
         );
-        assert!(
-            !exists<GasParameter>(@aptos_framework),
-            error::already_exists(EGAS_PARAMETER)
-        );
         move_to(aptos_framework, StateStorageUsage {
             epoch: 0,
             usage: Usage {
                 items: 0,
                 bytes: 0,
             }
         });
-        move_to(aptos_framework, GasParameter {
-            usage: Usage {
-                items: 0,
-                bytes: 0,
-            }
-        });
     }
 
     public(friend) fun on_new_block(epoch: u64) acquires StateStorageUsage {
+        assert!(
```

Comment:
Why add these checks now? If this fails, it crashes the entire network

Reply:
if the check doesn't fail, the next line will also fail, right?